classpath/java/io/BufferedInputStream.java

```diff
@@ -54,28 +59,7 @@ public int read(byte[] b, int offset, int length) throws IOException {
 
       count += remaining;
       position += remaining;
-      offset += remaining;
-      length -= remaining;
     }
-
-    while (length > 0) {
```

Comment:
With that loop the read operation will block until more data is available. Which may take a while. I think it's more efficient to return the available data. Or add a check before trying to read.